rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1032,6 +1032,61 @@ private void processReleaseRequests() {
                 activations
         );
 
+        // We have a BTC transaction, mark the UTXOs as spent and add the tx to the release set.
+        List<UTXO> availableUTXOs;
+        ReleaseTransactionSet releaseTransactionSet;
+        // Attempt access to storage first
+        // (any of these could fail and would invalidate both the tx build and utxo selection, so treat as atomic)
+        try {
+            availableUTXOs = getActiveFederationBtcUTXOs();
+            releaseTransactionSet = provider.getReleaseTransactionSet();
+        } catch (IOException exception) {
+            // Unexpected error accessing storage, log and fail
+            logger.error("Unexpected error accessing storage while attempting to processReleaseRequests ", exception);
+            return;
+        }
+
+        // Instead of creating a pegout transaction for every element in the release request queue,
+        // check if the elapsed time or blocks have gone by before creating the pegout transaction including all elements in the release request queue.
+        // Under a rush hour of peg-outs, the Bridge may need to create more than one peg-out transaction
+        // simultaneously per peg-out event to reduce the transaction size, and input count.
+        // Pending: Define the limit of the transaction size
+
+        if (activations.isActive(RSKIP271)) {
+            processReleasesInBatch(releaseRequestQueue, txBuilder, availableUTXOs, releaseTransactionSet);
+        } else {
+            processReleasesIndividually(releaseRequestQueue, txBuilder, availableUTXOs, releaseTransactionSet);
+        }
+    }
+
+    private void addPegoutTxToReleaseTransactionSet(BtcTransaction generatedTransaction,
+                                                 ReleaseTransactionSet releaseTransactionSet,
+                                                 ReleaseRequestQueue.Entry releaseRequest) {
+        if (activations.isActive(ConsensusRule.RSKIP146)) {
+            Keccak256 rskTxHash = releaseRequest.getRskTxHash();
+            // Add the TX
+            releaseTransactionSet.add(generatedTransaction, rskExecutionBlock.getNumber(), rskTxHash);
+            // For a short time period, there could be items in the release request queue that don't have the rskTxHash
+            // (these are releases created right before the consensus rule activation, that weren't processed before its activation)
+            // We shouldn't generate the event for those releases
+            if (rskTxHash != null) {
+                // Log the Release request
+                eventLogger.logReleaseBtcRequested(rskTxHash.getBytes(), generatedTransaction, releaseRequest.getAmount());
```

Comment:
Now that we are batching pegouts perhaps we should emit one `release_requested` event per release request. What do you think @kelvinator07 @josedahlquist ?